lib/ansible/galaxy/role.py

```diff
@@ -66,8 +66,18 @@ def __init__(self, galaxy, api, name, src=None, version=None, scm=None, path=Non
         self.scm = scm
 
         if path is not None:
-            if self.name not in path:
+            if not path.endswith(os.path.join(os.path.sep, self.name)):
                 path = os.path.join(path, self.name)
+            else:
+                # Look for a meta/main.ya?ml inside the potential role dir in case
```

Comment:
typo - main.yaml

Reply:
I was trying to be fancy and indicate that the `a` is option. I should probably just spell it out and say something like `meta/main.[yml|yaml]`.

Reply:
Oh, sorry for the density, that makes sense. :-)